lib/validate.js

```diff
@@ -1,82 +1,75 @@
-var merge = require('mout/object/merge')
+var jquery = require('./styles/jquery.js')
+var angular = require('./styles/angular.js')
+
+var limits = require('./utils/limits')
+var tickets = require('./utils/tickets')
+
+var defaults = require('./defaults.js')
+
 var semver = require('semver')
-var subjectExceptions = /^(fixup|squash)!|^\[[^\]]+\]:/
-var defaults = {
-  component: true,
-  components: [],
-  markerPattern: '^(clos|fix|resolv)(e[sd]|ing)',
-  actionPattern: '^([Cc]los|[Ff]ix|[Rr]esolv)(e[sd]|ing)\\s+[^\\s\\d]+(\\s|$)',
-  ticketPattern: '^(Closes|Fixes) (.*#|gh-|[A-Z]{2,}-)[0-9]+',
-  limits: {
-    subject: 72,
-    other: 80
-  }
-}
+var merge = require('mout/object/merge')
 
 module.exports = function (message, options) {
-  options = merge(defaults, options)
+  if (message === undefined) {
+    return ['Commit message is undefined, abort with error']
+  } else if (
+    !(typeof message === 'string' || message instanceof String)
+  ) {
+    return ['Commit message is not a string, abort with error']
+  } else if (message.length === 0) {
+    return ['Commit message is empty, abort with error']
+  }
+
+  if (options === undefined || options.style === undefined) {
+    options = merge(defaults.jquery, options)
+  } else if (options.style === 'jquery') {
```

Comment:
Could "merge" this condition into the previous one, since the result is the same:
```
if (options === undefined || options.style === undefined || options.style === 'jquery') {
 +    options = merge(defaults.jquery, options)
 +  } else if { ...
```